suite/spdz2k/src/main/java/dk/alexandra/fresco/suite/spdz2k/resource/storage/Spdz2kDummyDataSupplier.java

```diff
@@ -0,0 +1,73 @@
+package dk.alexandra.fresco.suite.spdz2k.resource.storage;
+
+import dk.alexandra.fresco.framework.util.ArithmeticDummyDataSupplier;
+import dk.alexandra.fresco.framework.util.MultiplicationTripleShares;
+import dk.alexandra.fresco.framework.util.Pair;
+import dk.alexandra.fresco.suite.spdz2k.datatypes.CompUInt;
+import dk.alexandra.fresco.suite.spdz2k.datatypes.CompUIntFactory;
+import dk.alexandra.fresco.suite.spdz2k.datatypes.Spdz2kInputMask;
+import dk.alexandra.fresco.suite.spdz2k.datatypes.Spdz2kSInt;
+import dk.alexandra.fresco.suite.spdz2k.datatypes.Spdz2kTriple;
+import java.math.BigInteger;
+
+public class Spdz2kDummyDataSupplier<
+    T extends CompUInt<?, ?, T>> implements
+    Spdz2kDataSupplier<T> {
+
+  private final int myId;
+  private final ArithmeticDummyDataSupplier supplier;
+  private final T secretSharedKey;
+  private final CompUIntFactory<T> factory;
+
+  public Spdz2kDummyDataSupplier(int myId, int noOfParties, T secretSharedKey,
+      CompUIntFactory<T> factory) {
+    this.myId = myId;
+    this.secretSharedKey = secretSharedKey;
+    this.factory = factory;
+    this.supplier = new ArithmeticDummyDataSupplier(myId, noOfParties,
+        BigInteger.ONE.shiftLeft(factory.getCompositeBitLength()));
+  }
+
+  @Override
+  public Spdz2kTriple<T> getNextTripleShares() {
+    MultiplicationTripleShares rawTriple = supplier.getMultiplicationTripleShares();
+    return new Spdz2kTriple<>(
+        toMarlinElement(rawTriple.getLeft()),
```

Comment:
`Marlin` -> `Spdz2k`  ... probably do some kind of search/replace over the project.